src/core/dbus-cgroup.c

```diff
@@ -1321,6 +1322,13 @@ int bus_cgroup_set_property(
                         if (r < 0)
                                 return r;
                         unit_write_drop_in_private(u, mode, name, buf);
+
+                        r = bpf_firewall_supported();
+                        if (r < 0)
+                                return r;
+                        if (r == 0)
+                                log_warning("Transient unit %s configures an IP firewall, but the local system does not support BPF/cgroup firewalling. "
+                                            "Proceeding WITHOUT firewalling in effect!", u->id);
```

Comment:
Perhaps this should cause the unit to fail instead? For example, a service which should only accept requests from localnet would be exposed to more hostile networks. Though it would mean that the unit file would need editing when used on hosts without BPF/cgroup firewalling.

Reply:
Well, the reason I explicitly chose not to make this fatal is that I'd like to ship all our services which do not require internet access (which are effectively all except for the journal remoting stuff), with either IPAddressDeny=any alone, or IPAddressDeny=any + IPAddressAllow=localhost, so that no IP communication is possible at all, or only with 127.0.0.1. Now, if we do set this, then we don't want to break the services completely on kernels that don't support eBPF/cgroup stuff, and thus I just added the loud warning.

Note that IPAddressDeny=any is in many ways a better option that PrivateNetwork= even though it has very similar effects. That's because PrivateNetwork= affects AF_NETLINK and AF_UNIX (at least the abstract namespace), and that's pretty awful, as it breaks libudev and such. IPAddressDeny= is hence a great option for everything that needs libudev but still wants to prohibit networking.